model/task_lifecycle.go

```diff
@@ -1080,30 +1080,43 @@ func UpdateBlockedDependencies(ctx context.Context, dependencies []task.Task) er
 }
 
 // UpdateUnblockedDependencies recursively marks all unattainable dependencies as attainable.
-func UpdateUnblockedDependencies(ctx context.Context, t *task.Task) error {
-	blockedTasks, err := t.FindAllMarkedUnattainableDependencies()
-	if err != nil {
-		return errors.Wrap(err, "getting dependencies marked unattainable")
+func UpdateUnblockedDependencies(ctx context.Context, dependencies []task.Task) error {
+	ctx, span := tracer.Start(ctx, "update-unblocked-dependencies")
+	defer span.End()
+
+	dependencyIDs := make([]string, 0, len(dependencies))
+	for _, dep := range dependencies {
+		dependencyIDs = append(dependencyIDs, dep.Id)
 	}
 
-	blockedTasks, err = task.MarkAllForUnattainableDependencies(ctx, blockedTasks, []string{t.Id}, false)
+	blockedTasks, err := task.FindAllDependencyTasksToModify(dependencies, true)
 	if err != nil {
-		return errors.Wrap(err, "marking attainable dependency for tasks")
+		return errors.Wrapf(err, "getting all tasks depending on tasks")
+	}
+	if len(blockedTasks) == 0 {
+		return nil
 	}
 
-	buildsToUpdate := make(map[string]bool)
-	for _, blockedTask := range blockedTasks {
-		if err := UpdateUnblockedDependencies(ctx, &blockedTask); err != nil {
-			return errors.WithStack(err)
-		}
+	blockedTasks, err = task.MarkAllForUnattainableDependencies(ctx, blockedTasks, dependencyIDs, false)
```

Comment:
nit: these tasks get unblocked by this update, should this var be called something else? Maybe `tasksToUnblock`?

Reply:
Agreed, done